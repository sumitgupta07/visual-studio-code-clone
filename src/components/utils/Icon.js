import React from 'react';
import './Icon.css';

const Icon = ({ label, onSelectIcon, iconClass }) => {
  return (
    <button onClick={() => onSelectIcon(label)}>
      <div className="tooltip">
        <div className="icon">{<i className={iconClass}></i>}</div>
        <div className="tooltiptext">
          <span>{label}</span>
        </div>
      </div>
    </button>
  );
};

export default Icon;
