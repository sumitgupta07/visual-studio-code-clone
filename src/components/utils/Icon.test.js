import React from 'react';
import Icon from './Icon';
import { shallow } from 'enzyme';

let handleSelectIcon;
let wrapper;
let mockProps;
beforeEach(() => {
  handleSelectIcon = jest.fn();
  mockProps = {
    label: 'Search',
    onSelectIcon: handleSelectIcon,
    iconClass: 'fas fa-search fa-rotate fa-3x'
  };
  wrapper = shallow(<Icon {...mockProps} />);
});

it('should render Icon component', () => {
  expect(wrapper).toMatchSnapshot();
});
it('should call onSelecIcon when button clicked', () => {
  wrapper.find('button').simulate('click');
  expect(handleSelectIcon).toHaveBeenCalled();
});
it('should pass "Search" as label on click', () => {
  wrapper.find('button').simulate('click');
  expect(handleSelectIcon).toHaveBeenCalledWith(mockProps.label);
});

it('should pass Icon class for Search as props', () => {
  expect(wrapper.find('i').hasClass(mockProps.iconClass)).toBe(true);
});
