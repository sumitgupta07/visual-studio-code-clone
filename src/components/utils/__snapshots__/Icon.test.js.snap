// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Icon component 1`] = `
<button
  onClick={[Function]}
>
  <div
    className="tooltip"
  >
    <div
      className="icon"
    >
      <i
        className="fas fa-search fa-rotate fa-3x"
      />
    </div>
    <div
      className="tooltiptext"
    >
      <span>
        Search
      </span>
    </div>
  </div>
</button>
`;
